designer/client/src/components/modals/Testing/TestingForm.tsx

```diff
@@ -0,0 +1,232 @@
+import { css, cx } from "@emotion/css";
+import loadable from "@loadable/component";
+import { Box, Button, FormGroup, FormLabel, Link, Typography } from "@mui/material";
+import type { WindowType } from "@touk/window-manager";
+import { flow, isEmpty } from "lodash";
+import React, { useCallback, useState } from "react";
+import { Trans, useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+
+import { testScenarioWithGeneratedData } from "../../../actions/nk/displayTestResults";
+import { getScenarioGraph, getTestCapabilities, getTestParameters } from "../../../reducers/selectors/graph";
+import { getFeatureSettings } from "../../../reducers/selectors/settings";
+import type { WindowKind } from "../../../windowManager";
+import type { ChangeableValue } from "../../ChangeableValue";
+import { CustomRadio } from "../../customRadio/CustomRadio";
+import { NodeInput } from "../../FormElements";
+import {
+    extendErrors,
+    getValidationErrorsForField,
+    literalIntegerValueValidator,
+    mandatoryValueValidator,
+    maximalNumberValidator,
+    minimalNumberValidator,
+} from "../../graph/node-modal/editors/Validators";
+import { NodeTable } from "../../graph/node-modal/NodeDetailsContent/NodeTable";
+import { nodeInput, nodeValue } from "../../graph/node-modal/NodeDetailsContent/NodeTableStyled";
+import { getProcessName } from "../../graph/node-modal/NodeDetailsContent/selectors";
+import ValidationLabels from "../../modals/ValidationLabels";
+import { AdhocTestingFormContext } from "../AdhocTesting/AdhocTestingFormContext";
+import { MarkdownForm } from "../AdhocTesting/MarkdownForm";
+import { useAdhocTestingAction } from "../AdhocTesting/useAdhocTestingAction";
+import { useAdhocTestingParametersValidation } from "../AdhocTesting/useAdhocTestingParametersValidation";
+import type { TestingData } from "./TestingDialog";
+
+export enum TestType {
+    withParameters = "withParameters",
+    withGeneratedData = "withGeneratedData",
+}
+
+export type FormValue = { testType: string };
+
+export type TouchedValue = Record<keyof FormValue, boolean>;
+
+interface TestingFormProps extends ChangeableValue<FormValue> {
+    handleSetTouched: (touched: TouchedValue) => void;
+    touched: TouchedValue;
+    testingData: WindowType<WindowKind, TestingData>;
+    closeDialog: () => void;
+}
+
+const DryRunTestingIcon = loadable(() => import("../../../assets/img/icons/test-dry-run.svg"));
+const GenerateAndTestIcon = loadable(() => import("../../../assets/img/icons/test-using-generated-data.svg"));
+
+export function TestingForm({ value, touched, onChange, handleSetTouched, testingData, closeDialog }: TestingFormProps): JSX.Element {
```

Comment:
IMO, this component makes way too much, we should separate it into a few components:

1. Scenario, which will have a logic to select a specific form, and after clicking into specific "sources" button will render a specific form
2. Testing Form, which will contain all ad-hoc related logic (buttons included here)
3. TestngFromSingleSource, rendered by the testing form
4. tetsingFromMultipleSourceWarning, rendered by the testing form
5. Live samples, which will contain testing from generated file logic (buttons included here)


Thanks to it, we will be able to hide logic related only to specific variants like useAdhocTesting* hooks, and we can remove redundant [enabled](https://github.com/TouK/nussknacker/pull/7779/files) flag from useAdhocTestingParametersValidation

Reply:
Separated into 
- TestingDialog - base for dialog that is opened 
- TestingForm - base content of the dialog, handles test type buttons
- TestVariantForm - chooses correct variant form
- and 3 variants:
  - TestWithParametersSingleSourceForm
  - TestWithParametersMultipleSourcesForm
  - TestWithGeneratedDataForm

Flag removed